File: day1/part2/solution.exs
defmodule Day1Part2 do
  def solve(input_path) do
    File.stream!(input_path)
    |> Enum.reduce({50, 0}, fn line, {current_position, zero_count} ->
      line = String.trim(line)
      if String.length(line) > 0 do
        {direction, distance_str} = String.split_at(line, 1)
        distance = String.to_integer(distance_str)

        new_zero_count =
          case direction do
            "R" ->
              zero_count + Integer.floor_div(current_position + distance, 100) - Integer.floor_div(current_position, 100)
            "L" ->
              zero_count + Integer.floor_div(current_position - 1, 100) - Integer.floor_div(current_position - distance - 1, 100)
          end

        new_position =
            case direction do
                "R" -> current_position + distance
                "L" -> current_position - distance
            end

        {new_position, new_zero_count}
      else
        {current_position, zero_count}
      end
    end)
    |> elem(1)
  end
end

password = Day1Part2.solve("day1/part2/input1.txt")
IO.puts("The password is: #{password}")

File: day1/part2/solution.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn solve(rotations: Vec<String>) -> i64 {
    let mut current_position: i64 = 50;
    let mut zero_count: i64 = 0;

    for line in rotations {
        if line.is_empty() {
            continue;
        }

        let direction = line.chars().next().unwrap();
        let distance: i64 = line[1..].parse().unwrap();

        if direction == 'R' {
            zero_count += (current_position + distance).div_euclid(100) - current_position.div_euclid(100);
            current_position += distance;
        } else if direction == 'L' {
            zero_count += (current_position - 1).div_euclid(100) - (current_position - distance - 1).div_euclid(100);
            current_position -= distance;
        }
    }

    zero_count
}

fn main() -> io::Result<()> {
    let path = Path::new("day1/part2/input1.txt");
    let file = File::open(&path)?;
    let lines = io::BufReader::new(file).lines();

    let rotations: Vec<String> = lines.map(|line| line.unwrap()).collect();
    
    let password = solve(rotations);
    println!("The password is: {}", password);

    Ok(())
}